### PR TITLE
Clean up Github Actions

### PR DESCRIPTION
**Pull request by BD103**
# Objective

- The Github Actions used by Bevy has grown organically and could use a bit of cleanup.
- There are a few inconsistencies and instances of repetitive code.

## Solution

- Go through each action and clean it up using the follow rules:
  - Don't use quotes when possible for strings
    - When required, always use single quotes
  - Use inline brackets for short lists (up to discretion)
  - Every action, job, and step should have a name
    - Names should uppercase first letter and acronyms only
    - Use American forms when possible (armor vs. armour, labeled vs. labelled)
  - Comments should be created generously as complete sentences ending in a period, when possible
  - Template syntax `${{}}` should surround contents with spaces (`${{ context }}` vs. `${{context}}`)

## Resources

- [Workflow syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions)
- [Creating a composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,16 +4,18 @@ on:
...
+  # Force colors, which can be displayed by Github.
   CARGO_TERM_COLOR: always
+  # Defaults to 'nightly', but can be modified in case the latest nightly is broken.
   NIGHTLY_TOOLCHAIN: nightly
 
+# If a new commit is pushed to the PR, cancel any old CI checks that might still be running.
 concurrency:
-  group: ${{github.workflow}}-${{github.ref}}
-  cancel-in-progress: ${{github.event_name == 'pull_request'}}
+  group: ${{ github.workflow }}-${{ github.ref }}
```
Comment (mgi388) [inline .github/workflows/ci.yml:17]: What are the chances of automating the “linting” that’s happening here? Eg what’s to stop someone coming back in a month and cleaning this up in reverse?

Comment (BD103) [reply .github/workflows/ci.yml:17]: Not much, in all honesty. There are linters that we can add that will force a consistent style, but the CI is already doing so much.

Comment (mgi388) [reply .github/workflows/ci.yml:17]: Yeah as far as YAML formatters that exist that I know of, it's either `prettier`, and I just found https://github.com/google/yamlfmt, which would arguably be easy and quick to install and run in CI compared to prettier.

Anyway, shame there's no amazing GH Actions tooling out there for formatting their files.

Comment (BD103) [reply .github/workflows/ci.yml:17]: Yeah. I use the [Github Actions Extension](https://marketplace.visualstudio.com/items?itemName=GitHub.vscode-github-actions) locally, but it doesn't do anything formatting-related.

**Comment by BD103** [comment]
I'm splitting off a few changes into a separate PR, closing this for now.

**Pull request closed without merging**